git_machete/client.py

```diff
@@ -1847,68 +1859,92 @@ def check_that_forkpoint_is_ancestor_or_equal_to_tip_of_branch(
-                        opt_as_root=False,
-                        opt_yes=True)
+        pr: Optional[GitHubPullRequest] = None
+        for pr in sorted(valid_prs, key=lambda x: x.number):
+            if pr.full_repository_name:
+                if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
+                    remote_already_added: Optional[str] = self.__get_added_remote_name_or_none(pr.repository_url)
+                    if not remote_already_added:
+                        remote_from_pr: str = pr.full_repository_name.split('/')[0]
+                        if remote_from_pr not in self.__git.get_remotes():
+                            self.__git.add_remote(remote_from_pr, pr.repository_url)
+                        remote_to_fetch: str = remote_from_pr
+                    else:
+                        remote_to_fetch = remote_already_added
+                    if remote != remote_to_fetch:
+                        if verbose:
+                            print(f"Fetching {remote_to_fetch}...")
+                        self.__git.fetch_remote(remote_to_fetch)
+                    if '/'.join([remote_to_fetch, pr.head]) not in self.__git.get_remote_branches():
+                        raise MacheteException(f"Could not check out PR #{pr.number} because its head branch `{pr.head}` is already deleted from `{remote_to_fetch}`.")
+            else:
+                warn(f'Pull request #{pr.number} comes from fork and its repository is already deleted. No remote tracking data will be set up for `{pr.head}` branch.')
+                if verbose:
+                    print(fmt(f"Checking out `{pr.head}` locally..."))
+                checkout_pr_refs(self.__git, remote, pr.number, LocalBranchShortName.of(pr.head))
+                self.flush_caches()
+            if pr.state == 'closed':
+                warn(f'Pull request #{pr.number} is already closed.')
+            debug('checkout_github_pr()', f'found {pr}')
+
+            path: List[LocalBranchShortName] = self.__get_path_from_pr_chain(pr, all_open_prs)
+            reversed_path: List[LocalBranchShortName] = path[::-1]  # need to add from root downwards
+            for index, branch in enumerate(reversed_path):
+                if branch not in self.managed_branches:
+                    if index == 0:
+                        self.add(
+                            branch=branch,
+                            opt_as_root=True,
+                            opt_onto=None,
+                            opt_yes=True,
+                            verbose=verbose)
+                    else:
+                        self.add(
+                            branch=branch,
+                            opt_onto=reversed_path[index - 1],
+                            opt_as_root=False,
+                            opt_yes=True,
+                            verbose=verbose)
+            print(fmt(f"Pull request `#{pr.number}` checked out at local branch `{pr.head}`"))
```

Comment:
Minor improvement to be considered:

```
$ git machete github checkout-prs --by=amalota
Pull request #407 checked out at local branch circle_ci_forbidden_access
Pull request #408 checked out at local branch update_contributing
Pull request #410 checked out at local branch ci/update_contributing_new_uid_gui_5
Pull request #416 checked out at local branch cli/checkout_prs_zsh_bash_completion
Branch 'docs/exclude_nix_link_from_remark' set up to track remote branch 'docs/exclude_nix_link_from_remark' from 'origin'.
Switched to a new branch 'docs/exclude_nix_link_from_remark'
Pull request #418 checked out at local branch docs/exclude_nix_link_from_remark
Branch 'ci/update_contributing_new_uid_gui_8' set up to track remote branch 'ci/update_contributing_new_uid_gui_8' from 'origin'.
Switched to a new branch 'ci/update_contributing_new_uid_gui_8'
Pull request #419 checked out at local branch ci/update_contributing_new_uid_gui_8
Pull request #421 checked out at local branch ci/circleci_inside_container_root
Pull request #422 checked out at local branch ci/circleci_inside_container_root_local
Branch 'docs/update_blog_docker' set up to track remote branch 'docs/update_blog_docker' from 'origin'.
Switched to a new branch 'docs/update_blog_docker'
Pull request #425 checked out at local branch docs/update_blog_docker
➜ 12:54 ~/git-machete docs/update_blog_docker PR #425 (amalota) virginia-dev:phoenix $ git machete github checkout-prs --by=amalota
Pull request #407 checked out at local branch circle_ci_forbidden_access
Pull request #408 checked out at local branch update_contributing
Pull request #410 checked out at local branch ci/update_contributing_new_uid_gui_5
Pull request #416 checked out at local branch cli/checkout_prs_zsh_bash_completion
Pull request #418 checked out at local branch docs/exclude_nix_link_from_remark
Pull request #419 checked out at local branch ci/update_contributing_new_uid_gui_8
Pull request #421 checked out at local branch ci/circleci_inside_container_root
Pull request #422 checked out at local branch ci/circleci_inside_container_root_local
Pull request #425 checked out at local branch docs/update_blog_docker
```

Reply:
...is that this message is only printed when the branch is actually first added to managed branches

Reply:
I dont understand, can you rephrase?

Reply:
The ```print(fmt(f"Pull request `#{pr.number}` checked out at local branch `{pr.head}`"))``` message was left intentionally so there is just minimal information on what's going on.

Reply:
Right... but the nuisance here is that it's printed for **all** matching branches every single time, even if they've been checked out locally & listed in .git/machete ("managed") previously!

Reply:
Ok, i understand now, i will try to implement this logic.